MarsRover/RoverClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MarsRover
+{
+    public class RoverClient
+    {
+        private Grid _grid;
+        private IHeading _heading;
+        private List<Obstacle> _obstacles;
+        private Rover _rover;
+        private List<RoverNavigationHistory> _navigationHistory = new List<RoverNavigationHistory>();
+
+
+        public RoverClient()
+        {
+            _grid = new Grid(3, 3, 5, 5);
+            _heading = new NorthHeading();
+            _obstacles = CreateObstacles(3);
+            _rover = new Rover(_grid, _heading, _obstacles);
+            AddToNavigationHistory();
+        }
+
+        
+        public void RoverInstructions()
+        {
+            var commands = string.Empty;
+            DisplayInstructions();
+            
+            while(!commands.Equals("X"))
+            {
+                Console.WriteLine($"Your current position on the surface grid is {_rover.XCoordinate()}, {_rover.YCoodinate()} and the Rover is facing {_rover.Direction()}.\n");
+                Console.WriteLine("Please enter your rover commands: ");
+                commands = Console.ReadLine().ToUpper().Replace(" ", "");
+                _rover = _rover.ExecuteCommands(commands);
+
+                AddToNavigationHistory();
+                if (_rover.ObstacleDetected()) { ObstacleAvoidance(); } 
+            }
+        }
+
+
+        public List<Obstacle> CreateObstacles(int numberOfObstacles)
```

Comment:
`IEnumerable`.  As a rule of thumb, always reach for `IEnumerable` first unless you need a concrete type like a `List`.